File: pyNA/src/noise_src_jl/oaspl.jl
function f_oaspl(x)
    
    # y = oaspl
    # x = spl
    
    # Compute 
    oaspl = 10*log10(sum(10 .^(x./10)))

    return oaspl
    
end


function f_oaspl!(oaspl, x)
    
    # y = oaspl
    # x = spl
    
    # Compute 
    oaspl .= 10*log10(sum(10 .^(x./10)))
    
end

File: old/src/noise_src_py/oaspl.py
import numpy as np

def oaspl(levels, spl: np.ndarray) -> np.ndarray:
    """
    Compute overall sound pressure level.

    :param levels: pyNA component computing noise levels
    :type levels: Levels
    :param spl: sound pressure level [dB]
    :type spl: np.ndarray

    :return: oaspl
    :rtype: np.ndarray

    """

    # Load options
    n_t = levels.options['n_t']

    # Compute OASPL by summing SPL logarithmically
    oaspl = np.zeros(n_t)
    for i in np.arange(n_t):
        oaspl[i] = 10 * np.log10(np.sum(10 ** (spl[i, :] / 10.)))

    return oaspl
